docs/sql-ref-syntax-ddl-drop-table.md

```diff
@@ -26,6 +26,8 @@ if the table is not `EXTERNAL` table. If the table is not present it throws an e
 
 In case of an external table, only the associated metadata information is removed from the metastore database.
 
+If the table is cached, the command uncaches the table and all its dependents.
```

Comment:
Can you double-check the behavior of DROP TABLE? Are the dependents (dataframe temp views) still valid?

Reply:
```sql
spark-sql> CREATE TABLE tbl (c0 INT);
spark-sql> INSERT INTO tbl SELECT 0;
spark-sql> CACHE TABLE tbl;
spark-sql> CREATE TEMP VIEW v1 AS SELECT c0 + 1 FROM tbl;
spark-sql> SELECT * FROM v1;
1
spark-sql> CACHE TABLE v1;
spark-sql> SELECT * FROM v1;
1
spark-sql> DROP TABLE tbl;
spark-sql> SELECT * FROM v1;
Error in query: Table or view not found: tbl; line 1 pos 19;
'Project [*]
+- 'SubqueryAlias v1
   +- View (`v1`, ['(c0 + 1)])
      +- 'Project [upcast('(c0 + 1), IntegerType) AS (c0 + 1)#156]
         +- 'Project [unresolvedalias(('c0 + 1), None)]
            +- 'UnresolvedRelation [tbl], [], false
```

Reply:
can you try with dataframe temp view?

Reply:
```scala
scala> sql("CREATE TABLE tbl (c0 INT)")
scala> sql("INSERT INTO tbl SELECT 0")
scala> val tbl = spark.table("tbl")
scala> tbl.cache()
scala> tbl.show(false)
+---+
|c0 |
+---+
|0  |
+---+
scala> tbl.select(($"c0" + 1).as("c1")).createOrReplaceTempView("tmp_view0")
scala> val v = spark.table("tmp_view0")
v: org.apache.spark.sql.DataFrame = [c1: int]

scala> v.cache()
res6: v.type = [c1: int]

scala> v.show(false)
+---+
|c1 |
+---+
|1  |
+---+
scala> sql("DROP TABLE tbl")
res8: org.apache.spark.sql.DataFrame = []

scala> v.show(false)
org.apache.hadoop.mapred.InvalidInputException: Input path does not exist: file:/Users/maximgekk/proj/doc-cmd-caching/spark-warehouse/tbl
```

Reply:
```scala
scala> Seq(0).toDF("c0").write.mode("overwrite").parquet("/Users/maximgekk/tmp/tbl")

scala> sql("CREATE TABLE tbl2 (c0 INT) USING parquet LOCATION '/Users/maximgekk/tmp/tbl'")
res11: org.apache.spark.sql.DataFrame = []

scala> val tbl2 = spark.table("tbl2")
tbl2: org.apache.spark.sql.DataFrame = [c0: int]

scala> tbl2.cache()
res12: tbl2.type = [c0: int]

scala> tbl2.show(false)
+---+
|c0 |
+---+
|0  |
+---+


scala> tbl2.select(($"c0" + 2).as("c1")).createOrReplaceTempView("tmp_view2")

scala> val v2 = spark.table("tmp_view2")
v2: org.apache.spark.sql.DataFrame = [c1: int]

scala> v2.cache()
res15: v2.type = [c1: int]

scala> v2.show(false)
+---+
|c1 |
+---+
|2  |
+---+


scala> sql("DROP TABLE tbl2")
res17: org.apache.spark.sql.DataFrame = []

scala> v2.show(false)
+---+
|c1 |
+---+
|2  |
+---+
```